### PR TITLE
WIP fix various bugs 

### PR DESCRIPTION
**Pull request by maraoz**
The package was using `Transaction#change` as a way of sending all input coins to an address. (See https://github.com/bitpay/bitcore/issues/1234)

requires: https://github.com/bitpay/bitcore/pull/1235

### CODE REVIEW

**Comment by maraoz** [comment]
This is ready. Need to merge https://github.com/bitpay/bitcore/pull/1235, release bitcore, and update package.json

**Comment by maraoz** [comment]
updated with new bitcore release, should be ready to merge :)

File: lib/transactions/payment.js [hunk 1]
```diff
@@ -25,32 +25,28 @@ function Payment(opts) {
   Transaction.call(this, opts.transaction);
 
   this.paymentAddress = new Address(opts.paymentAddress);
-  this._change = new Address(opts.changeAddress);
+  this.changeAddress = new Address(opts.changeAddress);
+  if (!this.outputs.length) {
+    this.change(this.changeAddress);
+    this._updateChangeOutput();
```
Comment (braydonf) [inline lib/transactions/payment.js]: Is this needed? `.change` calls `_updateChangeOutput()` https://github.com/bitpay/bitcore/blob/master/lib/transaction/transaction.js#L660

File: lib/transactions/commitment.js [hunk 2]
```diff
@@ -27,15 +25,17 @@ function Commitment(opts) {
   this.publicKeys = opts.publicKeys;
   this.outscript = Script.buildMultisigOut(this.publicKeys, 2);
   this.address = this.outscript.toScriptHashOut().toAddress();
-  this.change(this.address);
+  if (!this.outputs.length) {
```
Comment (braydonf) [inline lib/transactions/commitment.js]: The else branch here isn't tested.

File: lib/transactions/commitment.js [hunk 3]
```diff
@@ -27,15 +25,17 @@ function Commitment(opts) {
...
+    this.change(this.address);
+    this._updateChangeOutput();
```
Comment (braydonf) [inline lib/transactions/commitment.js]: `.change()` calls `_updateChangeOutput()`

**Comment by snakecharmer1024** [comment]
Thanks for this. Fixed a bug we were having :-$

**Comment by braydonf** [comment]
With comments addressed above (which should address coverage), I think this should be ready to merge, and remove the WIP.

**Pull request merged**